File: desktop-app/Southville8BEdgeUI/Models/Api/EventStatisticsDto.cs
using System.Text.Json.Serialization;

namespace Southville8BEdgeUI.Models.Api;

public class EventStatisticsDto
{
    [JsonPropertyName("totalEvents")]
    public int TotalEvents { get; set; }

    [JsonPropertyName("thisWeekEvents")]
    public int ThisWeekEvents { get; set; }

    [JsonPropertyName("upcomingEvents")]
    public int UpcomingEvents { get; set; }

    [JsonPropertyName("pastEvents")]
    public int PastEvents { get; set; }

    [JsonPropertyName("publishedEvents")]
    public int PublishedEvents { get; set; }

    [JsonPropertyName("draftEvents")]
    public int DraftEvents { get; set; }

    [JsonPropertyName("cancelledEvents")]
    public int CancelledEvents { get; set; }
}

File: core-api-layer/southville-nhs-school-portal-api-layer/src/events/dto/event-statistics.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class EventStatisticsDto {
  @ApiProperty({ description: 'Total number of events' })
  totalEvents: number;

  @ApiProperty({ description: 'Events created this week' })
  thisWeekEvents: number;

  @ApiProperty({ description: 'Upcoming published events' })
  upcomingEvents: number;

  @ApiProperty({ description: 'Past completed events' })
  pastEvents: number;

  @ApiProperty({ description: 'Published events count' })
  publishedEvents: number;

  @ApiProperty({ description: 'Draft events count' })
  draftEvents: number;

  @ApiProperty({ description: 'Cancelled events count' })
  cancelledEvents: number;
}
